src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,97 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.exception.EmptyTaskException;
+import duke.exception.InvalidDateException;
+import duke.exception.MissingDateException;
+
+/**
+ * Class representing a deadline task.
+ * A deadline task has a description containing the details of the task and
+ * a date by which the task is due.
+ */
+public class Deadline extends Task {
+    private final LocalDate dueDate;
+
+    private Deadline(String description, LocalDate dueDate) {
+        super(description);
+        this.dueDate = dueDate;
+    }
+
+    /**
+     * Creates a new Deadline object represented by the given command.
+     * The command is split into two parts by the " /by " delimiter, with the
+     * first being the task description and the second being the due date.
+     *
+     * @param task Description of task to be converted to a Deadline.
+     * @return Deadline object representing the given description.
+     * @throws EmptyTaskException If there is no text after "deadline ".
+     * @throws MissingDateException If " /by " is not given or if there is no date provided.
+     * @throws InvalidDateException If the date provided is in the wrong format.
+     */
+    public static Deadline create(String task)
+            throws EmptyTaskException, MissingDateException, InvalidDateException {
+        if (task.length() <= 9) {
+            throw new EmptyTaskException("deadline");
+        }
+
+        String[] taskInfo = task.substring(9).split(" /by ", 2);
+        if (taskInfo.length < 2) {
+            throw new MissingDateException();
+        }
+
+        LocalDate dateTime = null;
+        for (DateFormat format : DateFormat.values()) {
+            try {
+                dateTime = LocalDate.parse(taskInfo[1], format.toDateFormat());
+            } catch (DateTimeParseException ignored) {
+                continue;
+            }
+        }
```

Comment:
Great use of enhanced for-loop and enum to allow for multiple date formats input by the user 👍 
Did not think of this at all :)